### PR TITLE
Add TDD view to dashboard

### PR DESCRIPTION
**Pull request by djbarnwal**
This PR - 

1. Adds TDD View to the dashboard
2. Consolidates API queries into central stores

### Summary by CodeRabbit

- New Feature: Introduced an expanded view for individual measures in the dashboard, enhancing data visibility and analysis.
- New Feature: Added a search bar in the Time Detailed Dimension view, improving user navigation and data discovery.
- New Feature: Implemented a pivot table component for better data representation and interaction.
- New Feature: Enhanced comparison functionality in the dashboard, providing more detailed and flexible data comparison.
- Improvement: Optimized the structure and modularity of the Comparison Selector component, improving code maintainability.
- Improvement: Updated the Time Series Chart Container to support full-width display, enhancing the user interface.
- Improvement: Capitalized labels of time comparison options for better readability.

### CODE REVIEW

**Comment by ericpgreen2** [comment]
I've started taking some UXQA notes in [this Notion doc](https://www.notion.so/rilldata/Time-Dimension-Detail-UXQA-9bf47e7f5486411bb22815060864ab2e)

File: web-common/package.json [hunk 1]
```diff
@@ -56,5 +56,8 @@
     "vitest": "^0.31.0",
     "yaml": "^2.1.3"
   },
-  "type": "module"
+  "type": "module",
+  "dependencies": {
+    "regular-table": "^0.5.9"
```
Comment (ericpgreen2) [inline web-common/package.json:61]: This should be a devDependency

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
   },
   "dependencies": {
     "@bufbuild/protobuf": "^1.0.0",
+    "@faker-js/faker": "^8.0.2",
```
Comment (ericpgreen2) [inline package.json:32]: Can we remove this? I don't see it referenced anywhere

**Comment by ericpgreen2** [comment]
Can we remove the `tdd-virtualized-inactive` directory? Is that all old, unused code?

**Comment by ericpgreen2** [comment]
(Nit) There's a handful of places missing types which makes the code less approachable. Some I've run into:
- Functions in the `data-graphics` directory: `createSparkline`, `bisectData`, `createTimeFormat`
- Functions in `features/pivot`: `mergeBlocks`, `merge2DBlocks`
- Props in `TDDTable.svelte`

File: web-common/src/features/dashboards/time-series/multiple-dimension-queries.ts [hunk 3]
```diff
@@ -1,89 +1,258 @@
...
-  dimensionName: string,
-  selectedMeasureNames: string[],
-  filters: V1MetricsViewFilter,
-  start: string,
-  end: string,
-  interval: V1TimeGrain,
-  zone: string
-) {
-  if (!values && values.length == 0) return;
+export function getDimensionValuesForComparison(
```
[the remark this answers is not in the document]
Comment (djbarnwal) [reply web-common/src/features/dashboards/time-series/multiple-dimension-queries.ts:42]: This is actually used at 3 places with different specs. In overview chart with a limit to 7 values, in TDD chart with a limit to 11 values, and in the TDD table with a limit to 250 values. Splitting for overview and TDD would still retain a lot of if-else. Even if we split for Chart and table, that would also result in a similar number of if else due to the need of calling the toplist API when we have no selections.

File: web-common/src/features/dashboards/time-series/multiple-dimension-queries.ts [hunk 4]
```diff
@@ -1,89 +1,258 @@
...
+      const end = timeStore?.adjustedEnd;
+      const interval =
+        timeStore?.selectedTimeRange?.interval ?? timeStore?.minTimeGrain;
+      const zone = dashboardStore?.selectedTimezone;
+
+      if (!dimensionName) return;
+
+      return derived(
+        dimensionValues?.values?.map((value, i) => {
+          const updatedIncludeFilter = dimensionValues?.filter.include.map(
```
[the remark this answers is not in the document]
Comment (djbarnwal) [reply web-common/src/features/dashboards/time-series/multiple-dimension-queries.ts:186]: I am bit hesitant on creating another layer of `derived` stores. They would an additional layer of time complexity.

**Review by ericpgreen2 — APPROVED** [review]
High-level this is looking awesome. Though more to do, approving so you can continue working in smaller follow-on PRs (once tests pass).

**Pull request merged**